test/conformance/runtime_contract_types.go

```diff
@@ -16,25 +16,33 @@ limitations under the License.
 
 package conformance
 
+import (
+	"strconv"
+
+	"github.com/knative/serving/test"
+)
+
 //runtime_constract_types.go defines types that encapsulate run-time contract requirements as specified here: https://github.com/knative/serving/blob/master/docs/runtime-contract.md
 
-// ShouldEnvvars defines the environment variables that "SHOULD" be set.
-type ShouldEnvvars struct {
-	Service       string `json:"K_SERVICE"`
-	Configuration string `json:"K_CONFIGURATION"`
-	Revision      string `json:"K_REVISION"`
+// ShouldEnvVars defines environment variables that "SHOULD" be set.
+// To match these values with test service parameters,
+// map values must represent corresponding test.ResourceNames fields
+var ShouldEnvVars = map[string]string{
+	"K_SERVICE":       "Service",
+	"K_CONFIGURATION": "Config",
+	"K_REVISION":      "Revision",
 }
 
-// MustEnvvars defines environment variables that "MUST" be set.
-type MustEnvvars struct {
-	Port string `json:"PORT"`
+// MustEnvVars defines environment variables that "MUST" be set.
+var MustEnvVars = map[string]string{
+	"PORT": strconv.Itoa(test.EnvImageServerPort),
```

Comment:
I understand this didn't change with the replacement, but we should define the port as part of the service options to fetchRuntimeInfo so the port we are validating matches the port we specified.

We implicitly get the test case of system-defined ports as the runtime container is listening on $PORT so we would not receive a response back to parse if this variable was not set properly. So seems like validating the user-defined case makes the most sense as we could get back a valid port that the system is listening and sending traffic on, but that port could fail to be the one we defined.

Reply:
@dgerd that makes sense, added user-defined container port